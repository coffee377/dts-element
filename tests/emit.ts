import * as ts from 'typescript';
import * as dts from './index';

const top_level_element = dts.create_top_level_element({
  members: [
    dts.create_multi_line_comment({
      text: `hello\n\nworld`,
      align: true,
      prefix: '~ ',
    }),
    dts.create_class_declaration({
      name: 'AAA',
      extends: dts.create_general_type({
        name: 'SomethingElse',
      }),
    }),
  ],
});

it('should return correctly', () => {
  expect(JSON.stringify(dts.emit(top_level_element))).toMatchSnapshot();
});

it('should return correctly with removeComments', () => {
  expect(JSON.stringify(dts.emit(top_level_element, {
    removeComments: true,
  }))).toMatchSnapshot();
});

it('should return correctly with newLine', () => {
  expect(JSON.stringify(dts.emit(top_level_element, {
    newLine: ts.NewLineKind.CarriageReturnLineFeed,
  }))).toMatchSnapshot();
});

it('should return correctly with removeComments and newLine', () => {
  expect(JSON.stringify(dts.emit(top_level_element, {
    removeComments: true,
    newLine: ts.NewLineKind.CarriageReturnLineFeed,
  }))).toMatchSnapshot();
});
