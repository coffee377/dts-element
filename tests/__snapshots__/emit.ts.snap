// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return correctly 1`] = `"\\"/*~ hello\\\\n *~\\\\n *~ world\\\\n */\\\\nclass AAA extends SomethingElse {\\\\n}\\\\n\\""`;

exports[`should return correctly with newLine 1`] = `"\\"/*~ hello\\\\r\\\\n *~\\\\r\\\\n *~ world\\\\r\\\\n */\\\\r\\\\nclass AAA extends SomethingElse {\\\\r\\\\n}\\\\r\\\\n\\""`;

exports[`should return correctly with removeComments 1`] = `"\\"class AAA extends SomethingElse {\\\\n}\\\\n\\""`;

exports[`should return correctly with removeComments and newLine 1`] = `"\\"class AAA extends SomethingElse {\\\\r\\\\n}\\\\r\\\\n\\""`;
